### PR TITLE
[DESIGN-1109] Status icon component

### PR DESCRIPTION
**Pull request by mapsandapps**
[[DESIGN-1109](https://app.clickup.com/t/18033298/DESIGN-1109)]

If you have a better idea on the details, feel free to propose 'em. This seems to be working, at least, though.

### CODE REVIEW

**Comment by ryanjwilke** [comment]
Task linked: [DESIGN-1109 Status icons](https://app.clickup.com/t/18033298/DESIGN-1109)

File: components/StatusIcon/constants.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
+import { compileReactIconProperties } from '@cypress-design/react-icon'
```
Comment (elevatebart) [inline components/StatusIcon/constants.ts:1]: Importing react items in the common areas will bork up the typescript of TSX. You can experience it if you open a vue file and look at `class` or the same in react `className`.

Please move this in the react component.

File: components/StatusIcon/react/package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      "require": "./dist/index.umd.js"
+    }
+  },
+  "scripts": {
+    "build": "yarn build:module && yarn build:types",
+    "build:module": "rollup -c ./rollup.config.js",
+    "build:types": "tsc --project ./tsconfig.build.json"
+  },
+  "dependencies": {
+    "clsx": "*"
```
Comment (elevatebart) [inline components/StatusIcon/react/package.json:22]: missing dependencies:

- `react-icon`
- `icon-registry`

File: components/StatusIcon/vue/package.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    ".": {
+      "import": "./dist/index.es.js",
+      "require": "./dist/index.umd.js"
+    }
+  },
+  "scripts": {
+    "build": "yarn build:module && yarn build:types",
+    "build:module": "yarn vite build",
+    "build:types": "yarn vue-tsc --project ./tsconfig.build.json"
+  },
```
Comment (elevatebart) [inline components/StatusIcon/vue/package.json:20]: missing dependency to `icon-registry` at least

File: components/StatusIcon/vue/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "../../../tsconfig.vue.json",
+  "include": ["*.vue", "*.ts", "*.tsx", "../../../cypress/support/*.ts"]
```
Comment (elevatebart) [inline components/StatusIcon/vue/tsconfig.json:3]: add `../constants.ts`
```suggestion
  "include": ["*.vue", "*.ts", "*.tsx", "../../../cypress/support/*.ts", "../constants.ts"]
```

File: icon-registry/icons/cancelled-outline_x16.svg [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (elevatebart) [inline icon-registry/icons/cancelled-outline_x16.svg:1]: If you omit the width and height, the icons look way better in Mac Finder

```suggestion
<svg viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Screen Shot 2022-07-22 at 10 53 00 AM

**Review by elevatebart — CHANGES_REQUESTED** [review]
Real good PR,
I love the docs part!

Some small changes.

**Review by elevatebart — APPROVED** [review]

**Pull request merged**